src/unix/linux_like/linux/gnu/mod.rs

```diff
@@ -313,6 +313,55 @@ impl siginfo_t {
     }
 }
 
+cfg_if! {
+    if #[cfg(libc_union)] {
+        // Internal, for casts to access union fields
+        #[repr(C)]
+        #[derive(Copy,Clone)]
+        struct sifields_sigchld {
+            si_pid: ::pid_t,
+            si_uid: ::uid_t,
+            si_status: ::c_int,
+        }
+
+        // Internal, for casts to access union fields
+        #[repr(C)]
+        union sifields {
+            _align_pointer: *mut ::c_void,
+            sigchld: sifields_sigchld,
+        }
+
+        // Internal, for casts to access union fields. Note that some variants
+        // of sifields start with a pointer, which makes the alignment of
+        // sifields vary on 32-bit and 64-bit architectures.
+        #[repr(C)]
+        struct siginfo_f {
+            _si_signo: ::c_int,
+            _si_errno: ::c_int,
+            _si_code: ::c_int,
```

Comment:
These two fields may be swapped based on the original source code.
```c
#ifndef __ARCH_HAS_SWAPPED_SIGINFO
#define __SIGINFO 			\
struct {				\
	int si_signo;			\
	int si_errno;			\
	int si_code;			\
	union __sifields _sifields;	\
}
#else
#define __SIGINFO 			\
struct {				\
	int si_signo;			\
	int si_code;			\
	int si_errno;			\
	union __sifields _sifields;	\
}
#endif /* __ARCH_HAS_SWAPPED_SIGINFO */
```
Should we ignore that?

Reply:
They're the same size, so for the purposes of this cast it doesn't matter. If you'd like, I could change it to a slice of 3 c_int values to make that more explicit.

Reply:
Yes, they have the same size. But I am not talking about the size, I mean the C code have different orderings for different architectures but we don't have it. Shouldn't we have it or is it automatically solved?

Reply:
@pickfire We already have the different ordering for the actual base structure (prior to sifields); only MIPS sets `__ARCH_HAS_SWAPPED_SIGINFO`, and the MIPS targets already define the structure with the fields flipped. The structure defined *here* is never used to access the base fields, and just needs to skip past them, so their order doesn't matter; I'm suggesting that I can change them to `_base: [c_int; 3]` to make it explicit that the names don't matter.